src/compile.ts

```diff
@@ -407,6 +472,291 @@ export function compile(
+            );
+            // call the integration call function with the prewarm context and arguments
+            // At this point, we know native will not be undefined
+            // await integration.native.call(args, preWarmContext)
+            // FIXME: doesn't work without an array
+            // TODO: Support both sync and async function invocations: https://github.com/sam-goodwin/functionless/issues/105
+            return [
+              context.factory.createAwaitExpression(
+                context.factory.createCallExpression(
+                  context.factory.createPropertyAccessExpression(
+                    context.factory.createPropertyAccessExpression(
+                      node.expression,
+                      "native"
+                    ),
+                    "call"
+                  ),
+                  undefined,
+                  [
+                    context.factory.createArrayLiteralExpression(
+                      node.arguments
+                    ),
+                    // TODO: determine if this is needed at all?
+                    nativeExprContext.preWarmContext,
+                  ]
+                )
+              ),
+            ];
+          }
+        } else if (ts.isNewExpression(node)) {
+          const newType = checker.getTypeAtLocation(node);
+          // cannot create new resources in native runtime code.
+          const functionlessKind = getFunctionlessTypeKind(newType);
+          if (getFunctionlessTypeKind(newType)) {
+            throw Error(
+              `Cannot initialize new resources in a native function, found ${functionlessKind}.`
+            );
+          }
+          // TODO: check for CDK constructs
+          // const type = checker.getTypeAtLocation(node.expression);
+          // if (isCDKConstruct(type)) {
+          //   throw Error(
+          //     `Cannot initialize new resources in a native function, found ${type.symbol.name}.`
+          //   );
+          // }
+        } else if (ts.isReturnStatement(node)) {
+          const child = node.expression
+            ? toNativeExpr(node.expression, context, nativeExprContext)
+            : undefined;
+          if (!child) {
+            return node;
+          } else if (Array.isArray(child)) {
+            // if multiple nodes are returned, assume the last one is the value to assign and the rest should be injected as siblings.
+            const [last, ...rest] = child.reverse();
+            return [
+              ...rest.reverse(),
+              context.factory.updateReturnStatement(
+                node,
+                last as ts.Expression
+              ),
+            ];
+          } else {
+            return context.factory.updateReturnStatement(
+              node,
+              child as ts.Expression
+            );
+          }
+        } else if (ts.isVariableDeclarationList(node)) {
+          // bubble any extra nodes to the top of the variable declaration list.
```

Comment:
why? I don't understand.

Reply:
See above, will remove anyways, but it was fun... is there a better way to do this?
https://github.com/sam-goodwin/functionless/pull/103#discussion_r867284322